module1/homework/make_unique/make_unique.hpp

```diff
@@ -0,0 +1,14 @@
+#include <memory>
+
+namespace  cs {
+
+    template<typename T, typename... Args, typename = std::enable_if_t<!std::is_array_v<T>>>
+    std::unique_ptr <T> make_unique(Args &&... args) {
+        return std::unique_ptr<T>{new T{std::forward<Args>(args)...}};
+    }
+
+    template<typename T>
+    std::unique_ptr <T> make_unique(size_t size) {
+        return std::unique_ptr<T>{new std::remove_extent_t<T>[size] {}};
+    }
```

Comment:
This should be a proper implementation. Without that memory leaks / undefined behavior is possible.
```suggestion
    template<typename T>
    std::unique_ptr<T[]> make_unique(size_t size) {
        return std::unique_ptr<T[]>{new std::remove_extent_t<T>[size] {}};
    }
```

Reply:
Oh, ok

Reply:
However, this is not compiling :D And local valgrind without it showed no leaks